src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -487,7 +532,7 @@ static int parse_argv(int argc, char *argv[]) {
                 { "help",     no_argument,       NULL, 'h'          },
                 { "version",  no_argument,       NULL, ARG_VERSION  },
                 { "list",     no_argument,       NULL, ARG_LIST     },
-                { "query",    no_argument,       NULL, ARG_QUERY    },
+                { "query",    optional_argument, NULL, ARG_QUERY    },
```

Comment:
please do not add new uses of "optional_argument" to our tree. It's quite problematic since it makes things such as "--foo bar" ambiguous, it's not clear if this is --foo with an argument of bar, or if this is a no-argument --foo with a later positional argument of "bar". in particular, it's a really bad idea to change things for existing arguments, since that creates a compat problem: command lines that did one thing before now do a completely different thing, potentially.

Reply:
or in other words, just add a new switch, maybe --filter-pid= or so

Reply:
--filter was added